rpc/src/cache_block_meta_service.rs

```diff
@@ -30,40 +30,39 @@ impl CacheBlockMetaService {
     ) -> Self {
         let thread_hdl = Builder::new()
             .name("solCacheBlkTime".to_string())
-            .spawn(move || loop {
-                if exit.load(Ordering::Relaxed) {
-                    break;
-                }
-                let recv_result = cache_block_meta_receiver.recv_timeout(Duration::from_secs(1));
-                match recv_result {
-                    Err(RecvTimeoutError::Disconnected) => {
+            .spawn(move || {
+                info!("CacheBlockMetaService has started");
+                loop {
+                    if exit.load(Ordering::Relaxed) {
                         break;
                     }
-                    Ok(bank) => {
-                        let mut cache_block_meta_timer = Measure::start("cache_block_meta_timer");
-                        Self::cache_block_meta(&bank, &blockstore);
-                        cache_block_meta_timer.stop();
-                        if cache_block_meta_timer.as_ms() > CACHE_BLOCK_TIME_WARNING_MS {
-                            warn!(
-                                "cache_block_meta operation took: {}ms",
-                                cache_block_meta_timer.as_ms()
-                            );
+
+                    let bank = match cache_block_meta_receiver.recv_timeout(Duration::from_secs(1))
+                    {
+                        Ok(bank) => bank,
+                        Err(RecvTimeoutError::Timeout) => continue,
+                        Err(err @ RecvTimeoutError::Disconnected) => {
+                            info!("CacheBlockMetaService is stopping because: {err}");
+                            break;
                         }
+                    };
+
+                    if let Err(err) = Self::cache_block_meta(&bank, &blockstore) {
+                        error!("CacheBlockMetaService is stopping because: {err}");
+                        // Set the exit flag to allow other services to gracefully stop
+                        exit.store(true, Ordering::Relaxed);
+                        break;
                     }
-                    _ => {}
                 }
+                info!("CacheBlockMetaService has stopped");
             })
             .unwrap();
         Self { thread_hdl }
     }
 
-    fn cache_block_meta(bank: &Bank, blockstore: &Blockstore) {
-        if let Err(e) = blockstore.cache_block_time(bank.slot(), bank.clock().unix_timestamp) {
-            error!("cache_block_time failed: slot {:?} {:?}", bank.slot(), e);
-        }
-        if let Err(e) = blockstore.cache_block_height(bank.slot(), bank.block_height()) {
-            error!("cache_block_height failed: slot {:?} {:?}", bank.slot(), e);
-        }
+    fn cache_block_meta(bank: &Bank, blockstore: &Blockstore) -> Result<(), BlockstoreError> {
+        blockstore.cache_block_time(bank.slot(), bank.clock().unix_timestamp)?;
+        blockstore.cache_block_height(bank.slot(), bank.block_height())
```

Comment:
does it matter that we're losing observability for which insertion (block time vs. block height) failed?

Reply:
My opinion is no; there is nothing really special happening in either of these columns. More so, rocksdb errors are typically "sticky". So, if the first one fails, I expect the second one (and all other writes) to fail. So, while this thread will bail out, I imagine `ReplayStage` or `WindowService` will have panicked somewhere

Reply:
yeah, that's my understanding as well